tests/simulationbase.py

```diff
@@ -173,55 +182,112 @@ def _simulate(self, task_params: dict, tmpdir, expected_frames: list):
             time.sleep(3)
 
             task_id = 'test_task_id123'
-            req_task_work_dir = req_work_dir / task_id
-            req_task_work_dir.mkdir()
-            req_task_resources_dir = req_task_work_dir / constants.RESOURCES_DIR
-            req_task_resources_dir.mkdir()
-            req_task_net_resources_dir = \
-                req_task_work_dir / constants.NETWORK_RESOURCES_DIR
-            req_task_net_resources_dir.mkdir()
-            req_task_results_dir = req_task_work_dir / constants.RESULTS_DIR
-            req_task_results_dir.mkdir()
-            req_task_net_results_dir = \
-                req_task_work_dir / constants.NETWORK_RESULTS_DIR
-            req_task_net_results_dir.mkdir()
-
-            prov_task_work_dir = prov_work_dir / task_id
-            prov_task_work_dir.mkdir()
-            prov_task_net_resources_dir = \
-                prov_task_work_dir / constants.NETWORK_RESOURCES_DIR
-            prov_task_net_resources_dir.mkdir()
+            req_task_work_dir, req_task_resources_dir, \
+                req_task_net_resources_dir, req_task_results_dir, \
+                req_task_net_results_dir = \
+                self._make_req_dirs(req_work_dir, task_id)
+
+            prov_task_work_dir, prov_task_net_resources_dir = \
+                self._make_prov_dirs(prov_work_dir, task_id)
 
             self._put_cube_to_resources(req_task_resources_dir)
 
             self._create_task(requestor, task_id, task_params)
 
             for _ in range(task_params['subtasks_count']):
 
-                subtask_id, subtask_params = \
-                    self._get_next_subtask(requestor, task_id)
-                assert subtask_params['resources'] == [0]
-
-                self._copy_resources_from_requestor(
+                self._do_subtask(
+                    task_id,
+                    requestor,
+                    provider,
                     req_task_net_resources_dir,
+                    req_task_net_results_dir,
                     prov_task_net_resources_dir,
-                    subtask_params,
+                    prov_task_work_dir
                 )
-                prov_subtask_work_dir = prov_task_work_dir / subtask_id
-                prov_subtask_work_dir.mkdir()
-                self._compute(provider, task_id, subtask_id, subtask_params)
-                self._copy_result_from_provider(
-                    prov_subtask_work_dir,
+
+            for frame in expected_frames:
+                filename = f'result{frame:04d}.{task_params["format"]}'
+                result_file = req_task_results_dir / filename
+                assert result_file.exists()
+        finally:
+            self._close_server(requestor_server)
+            self._close_server(provider_server)
+
+
+    def _simulate_restart(self, task_params: dict, tmpdir,
```

Comment:
Tbh I dunno if that should be called `restart`, it doesn't really restart anything, does it?
This effectively checks that if we change the name of the task directory between subtasks, will it still work?
Which imo could be implemented just as such.

Reply:
its the way the golem-app would restart a task when the task stops after total - 1 subtasks.

It indeed boils down to copy-ing the whole folder to a new name and starting the task again. But its the first step we can expand on when we start actually restarting from the golem-app

Reply:
> But its the first step we can expand on when we start actually restarting from the golem-app

What do you mean by that? How do you see it being expanded?

Reply:
i do not know yet, but for example: By having a failed subtask, or forcing a completed one to be re-computed

Reply:
Imo these are cases for the `discard` command, which doesn't have to be conceptually connected to `restart` at all.